ipam/allocator.go

```diff
@@ -46,9 +77,11 @@ type Allocator struct {
 // NewAllocator returns an instance of libnetwork ipam
 func NewAllocator(ds datastore.DataStore) (*Allocator, error) {
 	a := &Allocator{}
-	a.subnets = make(map[subnetKey]*SubnetInfo)
+	a.subnets = make(map[subnetKey]*PoolData)
 	a.addresses = make(map[subnetKey]*bitseq.Handle)
-	a.internalHostSize = defaultInternalHostSize
+	a.predefined = make(map[string][]*net.IPNet, 2)
+	a.predefined[localAddressSpace] = localPredefinedPools
+	a.predefined[globalAddressSpace] = globalPredefinedPools
 	a.store = ds
 	a.App = "ipam"
 	a.ID = dsConfigKey
```

Comment:
I see now where the confusion arises.
There is no need for the `ID` field in `Allocator` given we will only have one allocator instance.
`App` itself can be a constant. 

I will rework this part.

I will also consider changing the ipam config storing from a single blob to separated ones per address space as you were suggesting.

Thanks.